TYPO3.Flow/Classes/TYPO3/Flow/Resource/Storage/WritableFileSystemStorage.php

```diff
@@ -162,10 +162,10 @@ protected function moveTemporaryFileToFinalDestination($temporaryFile, $finalTar
         if (!file_exists(dirname($finalTargetPathAndFilename))) {
             Files::createDirectoryRecursively(dirname($finalTargetPathAndFilename));
         }
-        if (rename($temporaryFile, $finalTargetPathAndFilename) === false) {
-            unlink($temporaryFile);
+        if (copy($temporaryFile, $finalTargetPathAndFilename) === false) {
             throw new StorageException(sprintf('The temporary file of the file import could not be moved to the final target "%s".', $finalTargetPathAndFilename), 1381156103);
         }
+        unlink($temporaryFile);
```

Comment:
The problem with copy and unlink is that it's not atomic. But should be fine in this case.